# A-Cost of pollution

In [1]:
import pandas as pd

In [2]:
dfcost = pd.read_csv("Cost of pollution EU (2021)_update2.xlsx - Cost of pollution EU (2021).csv")

In [3]:
dfcost.head()

Facility_INSPIRE_ID  \
0                           PL.MŚ/000000349.FACILITY   
1  https://registry.gdi-de.org/id/de.nw.inspire.p...   
2  https://registry.gdi-de.org/id/de.bb.inspire.p...   
3  https://registry.gdi-de.org/id/de.sn.sax4inspi...   
4  https://registry.gdi-de.org/id/de.nw.inspire.p...   

                                        facilityName          city  \
0  PGE Górnictwo i Energetyka Konwencjonalna S.A....      Rogowiec   
1                   RWE Power AG - Kraftwerk Neurath  Grevenbroich   
2                        LEAG, Kraftwerk Jänschwalde     Teichland   
3                                  Kraftwerk Boxberg  Boxberg/O.L.   
4                 RWE Power AG Kraftwerk Niederaußem      Bergheim   

                region  Country      sector_en  \
0  województwo łódzkie   Poland  Energy sector   
1  Nordrhein-Westfalen  Germany  Energy sector   
2          Brandenburg  Germany  Energy sector   
3              Sachsen  Germany  Energy sector   
4  Nordrhein-Westfalen  Germany  Energy sector   

                                         activity_en  \
0  Thermal power stations and other combustion in...   
1  Thermal power stations and other combustion in...   
2  Thermal power stations and other combustion in...   
3  Thermal power stations and other combustion in...   
4  Thermal power stations and other combustion in...   

                                 activity_translated      cost_2021  
0  Elektrociepłownie i inne instalacje do spalani...  5.618.554.368  
1     Wärmekraftwerke und andere Verbrennungsanlagen  3.109.096.192  
2     Wärmekraftwerke und andere Verbrennungsanlagen  2.674.206.976  
3     Wärmekraftwerke und andere Verbrennungsanlagen  2.500.956.160  
4     Wärmekraftwerke und andere Verbrennungsanlagen  2.427.463.680

## 1) Cleaning
Dropping mostly useless columns

In [4]:
dfcost = dfcost.drop(["Facility_INSPIRE_ID", "activity_translated"], axis='columns') 

### Cleaning the cost column
Originally done this in the BE df. Pushing code up so it cleans the global df

In [5]:
dfcost["cost_2021"] = dfcost["cost_2021"].str.replace('.', '') 
dfcost["cost_2021"] = dfcost["cost_2021"].fillna(0)
dfcost["cost_2021"] = dfcost["cost_2021"].astype(int)

## 2) Only Belgium 
Keeping original df for future comparison

In [6]:
dfcostbe = dfcost.loc[dfcost["Country"] == "Belgium"]

In [7]:
dfcostbe.head()

facilityName       city             region  Country  \
21       ARCELORMITTAL BELGIUM - GENT       Gent  Flandre orientale  Belgium   
30     TotalEnergies Refinery Antwerp  Antwerpen             Anvers  Belgium   
34          AHLSTROM  MUNKSJÖ MALMEDY    Malmedy              Liège  Belgium   
63  Electrabel - Centrale Knippegroen       Gent  Flandre orientale  Belgium   
74                     BASF ANTWERPEN  Antwerpen             Anvers  Belgium   

                                   sector_en  \
21       Production and processing of metals   
30                             Energy sector   
34  Paper and wood production and processing   
63                             Energy sector   
74                         Chemical industry   

                                          activity_en  cost_2021  
21  Installations for the production of pig iron o...  983884928  
30                     Mineral oil and gas refineries  794067904  
34  Industrial plants for the production of paper ...  694007232  
63  Thermal power stations and other combustion in...  429914944  
74  Chemical installations for the production on a...  392816128

In [8]:
dfcostbe.to_csv("pollution_cost_be.csv", index=False)

## Per territory

In [9]:
dfcostbe.region.value_counts().sort_values(ascending=False)  #Regions are wrong, would require manual cleaning. 

region
Anvers                 55
Flandre-Occidentale    38
Région wallonne        29
Région flamande        26
Flandre orientale      15
Limbourg               14
Liège                  11
Hainaut                 9
Luxembourg              6
Brabant flamand         6
Bruxelles-Capitale      3
Province de Namur       1
Brabant wallon          1
Name: count, dtype: int64

In [10]:
dfcostbe.groupby("city")["cost_2021"].sum().sort_values(ascending=False)  #Cities make sense

city
Antwerpen                2226697179
Gent                     1859324153
Malmedy                   694007232
Obourg                    269452000
Gaurain-Ramecroix         235492576
                            ...    
Langemark-Poelkapelle        694604
Ledegem                      687726
Tielt                        634237
Vaux-Sous-Chèvremont          57171
Ieper                          5277
Name: cost_2021, Length: 125, dtype: int64

In [11]:
dfcitycostbe = dfcostbe.groupby("city")["cost_2021"].sum().sort_values(ascending=False)

In [12]:
dfcitycostbe.to_csv("pollution_cost_be_per_city.csv")

## Analysis

In [13]:
dfcostbe.cost_2021.sum()

7996066490

In [14]:
dfcostbe.describe()

cost_2021
count  2.140000e+02
mean   3.736480e+07
std    1.121150e+08
min    5.277000e+03
25%    8.530872e+05
50%    2.354905e+06
75%    2.757097e+07
max    9.838849e+08

Notes:
214 facilities cost > 7,996 million
The data shows a wide range of costs, from a minimum of 5,277 to a maximum of nearly 1 billion.
The mean and median are quite different (mean is much higher than the median), suggesting that the data is skewed to the right (a few very high costs are pulling the mean up).
The large standard deviation relative to the mean further indicates high variability in the costs.

In [15]:
dfcostbe[dfcostbe["cost_2021"] > 27570969]["cost_2021"].sum()


7212190236

In [16]:
dfcostbe[dfcostbe["cost_2021"] > 27570969].count()

facilityName    54
city            53
region          54
Country         54
sector_en       54
activity_en     54
cost_2021       54
dtype: int64

Conclusion: 54 facilities are responsible for 90% of pollution costs 

In [17]:
# Doubling value
dfcostbe[dfcostbe["cost_2021"] > 55141938]["cost_2021"].sum()

6492573940

In [18]:
dfcostbe[dfcostbe["cost_2021"] > 55141938].count()

facilityName    33
city            33
region          33
Country         33
sector_en       33
activity_en     33
cost_2021       33
dtype: int64

Conclusion: 33 facilities are responsible for 81% of pollution costs 

In [19]:
dfcostbe.sort_values(by="cost_2021", ascending=False).head(30)

facilityName  \
21                        ARCELORMITTAL BELGIUM - GENT   
30                      TotalEnergies Refinery Antwerp   
34                           AHLSTROM  MUNKSJÖ MALMEDY   
63                   Electrabel - Centrale Knippegroen   
74                                      BASF ANTWERPEN   
104  EXXONMOBIL PETROLEUM & CHEMICAL - ESSO RAFFINA...   
125                   HOLCIM Belgique - Usine d'OBOURG   
133                     ELECTRABEL CENTRALE RODENHUIZE   
145                    CCB - Site de Gaurain-Ramecroix   
194                            CBR sa - Site d'Antoing   
219                             CBR sa - Site de Lixhe   
221                                     BURGO ARDENNES   
327                                  UMICORE - HOBOKEN   
395                      TotalEnergies Olefins Antwerp   
426                  ELECTRABEL - CENTRALE D'AMERCOEUR   
505                         AIR LIQUIDE LARGE INDUSTRY   
541             INDUSTRIE DU BOIS VIELSALM & CIE - IBV   
546                                              ARGEX   
550                                   EVONIK ANTWERPEN   
557                     ZANDVLIET POWER - TERREIN BASF   
584                            STORA ENSO LANGERBRUGGE   
586                                 EUROCHEM ANTWERPEN   
628                                           BIOWANZE   
652                                            LUMINUS   
662                                            T POWER   
687          CARRIERES ET FOURS A CHAUX DUMONT WAUTIER   
697                                            LANXESS   
715          ELECTRABEL - CENTRALE BAUDOUR/St GHISLAIN   
725                       INOVYN Manufacturing Belgium   
728                              TESSENDERLO GROUP HAM   

                        city             region  Country  \
21                      Gent  Flandre orientale  Belgium   
30                 Antwerpen             Anvers  Belgium   
34                   Malmedy              Liège  Belgium   
63                      Gent  Flandre orientale  Belgium   
74                 Antwerpen             Anvers  Belgium   
104                Antwerpen             Anvers  Belgium   
125                   Obourg    Région wallonne  Belgium   
133                     Gent  Flandre orientale  Belgium   
145        Gaurain-Ramecroix    Région wallonne  Belgium   
194                  Antoing    Région wallonne  Belgium   
219                    Lixhe    Région wallonne  Belgium   
221                   Virton         Luxembourg  Belgium   
327                Antwerpen             Anvers  Belgium   
395                Antwerpen             Anvers  Belgium   
426                     Roux    Région wallonne  Belgium   
505                Antwerpen             Anvers  Belgium   
541                 Vielsalm         Luxembourg  Belgium   
546              Zwijndrecht             Anvers  Belgium   
550                Antwerpen             Anvers  Belgium   
557                Antwerpen             Anvers  Belgium   
584                     Gent  Flandre orientale  Belgium   
586                Antwerpen             Anvers  Belgium   
628                    Wanze    Région wallonne  Belgium   
652                     Gent  Flandre orientale  Belgium   
662              Tessenderlo           Limbourg  Belgium   
687  Saint-Georges-Sur-Meuse    Région wallonne  Belgium   
697                Antwerpen             Anvers  Belgium   
715                  Baudour         Luxembourg  Belgium   
725       Jemeppe-Sur-Sambre  Province de Namur  Belgium   
728                      Ham           Limbourg  Belgium   

                                    sector_en  \
21        Production and processing of metals   
30                              Energy sector   
34   Paper and wood production and processing   
63                              Energy sector   
74                          Chemical industry   
104                             Energy sector   
125                          Mineral in

In [20]:
dfcostbe.groupby("facilityName")["cost_2021"].sum().sort_values(ascending=False).head(30) #Head can be changed to include more or less

facilityName
ARCELORMITTAL BELGIUM - GENT                           983884928
TotalEnergies Refinery Antwerp                         794067904
AHLSTROM  MUNKSJÖ MALMEDY                              694007232
Electrabel - Centrale Knippegroen                      429914944
BASF ANTWERPEN                                         392816128
EXXONMOBIL PETROLEUM & CHEMICAL - ESSO RAFFINADERIJ    315024288
HOLCIM Belgique - Usine d'OBOURG                       269452000
ELECTRABEL CENTRALE RODENHUIZE                         255204144
CCB - Site de Gaurain-Ramecroix                        235492576
CBR sa - Site d'Antoing                                195195872
CBR sa - Site de Lixhe                                 180051568
BURGO ARDENNES                                         178010832
UMICORE - HOBOKEN                                      127319424
AIR LIQUIDE LARGE INDUSTRY                             117401352
TotalEnergies Olefins Antwerp                          107426384
ELECTRABEL -

In [21]:
dfcostbe.groupby("facilityName")["cost_2021"].sum().sort_values(ascending=False).head(30).sum()

6369953881

Conclusion: 30 facilities are responsible for 80% (79.66%) of pollution costs

### Per territory and sector
Preparing data for visualisation

In [22]:
top30 = dfcostbe.sort_values(by="cost_2021", ascending=False).head(30)

In [23]:
#1.Metals
top30[top30["sector_en"] == "Production and processing of metals"]

facilityName       city             region  Country  \
21   ARCELORMITTAL BELGIUM - GENT       Gent  Flandre orientale  Belgium   
327             UMICORE - HOBOKEN  Antwerpen             Anvers  Belgium   

                               sector_en  \
21   Production and processing of metals   
327  Production and processing of metals   

                                           activity_en  cost_2021  
21   Installations for the production of pig iron o...  983884928  
327  Installation for the production of non-ferrous...  127319424

In [24]:
#2. Energy
top30[top30["sector_en"] == "Energy sector"]

facilityName         city  \
30                      TotalEnergies Refinery Antwerp    Antwerpen   
63                   Electrabel - Centrale Knippegroen         Gent   
104  EXXONMOBIL PETROLEUM & CHEMICAL - ESSO RAFFINA...    Antwerpen   
133                     ELECTRABEL CENTRALE RODENHUIZE         Gent   
426                  ELECTRABEL - CENTRALE D'AMERCOEUR         Roux   
541             INDUSTRIE DU BOIS VIELSALM & CIE - IBV     Vielsalm   
557                     ZANDVLIET POWER - TERREIN BASF    Antwerpen   
652                                            LUMINUS         Gent   
662                                            T POWER  Tessenderlo   
715          ELECTRABEL - CENTRALE BAUDOUR/St GHISLAIN      Baudour   

                region  Country      sector_en  \
30              Anvers  Belgium  Energy sector   
63   Flandre orientale  Belgium  Energy sector   
104             Anvers  Belgium  Energy sector   
133  Flandre orientale  Belgium  Energy sector   
426    Région wallonne  Belgium  Energy sector   
541         Luxembourg  Belgium  Energy sector   
557             Anvers  Belgium  Energy sector   
652  Flandre orientale  Belgium  Energy sector   
662           Limbourg  Belgium  Energy sector   
715         Luxembourg  Belgium  Energy sector   

                                           activity_en  cost_2021  
30                      Mineral oil and gas refineries  794067904  
63   Thermal power stations and other combustion in...  429914944  
104                     Mineral oil and gas refineries  315024288  
133  Thermal power stations and other combustion in...  255204144  
426  Thermal power stations and other combustion in...  101475648  
541  Thermal power stations and other combustion in...   81751544  
557  Thermal power stations and other combustion in...   79335072  
652  Thermal power stations and other combustion in...   67063012  
662  Thermal power stations and other combustion in...   65962304  
715  Thermal power stations and other combustion in...   59909400

(I left this part of the analysis on standby)

# B- Belgium's pollution over the years

In [25]:
dftime = pd.read_csv("Belgium.xlsx - Belgium.csv")

In [26]:
dftime.head()

Facility_INSPIRE_ID       facilityName  \
0  BE.BRU/100010004.FACILITY  BRUXELLES ENERGIE   
1  BE.BRU/100010004.FACILITY  BRUXELLES ENERGIE   
2  BE.BRU/100010006.FACILITY      AUDI BRUSSELS   
3  BE.BRU/100010015.FACILITY    STEP SUD (SBGE)   
4  BE.BRU/100010015.FACILITY    STEP SUD (SBGE)   

                                   parentCompanyName       city  \
0  Bruxelles Propreté - Agence Régionale pour la ...  Bruxelles   
1  Bruxelles Propreté - Agence Régionale pour la ...  Bruxelles   
2                                            Audi AG     Forest   
3                                               SBGE     Forest   
4                                               SBGE     Forest   

               region                                   sector  \
0  Bruxelles-Capitale    Gestion des déchets et des eaux usées   
1  Bruxelles-Capitale    Gestion des déchets et des eaux usées   
2  Bruxelles-Capitale  Production et transformation des métaux   
3  Bruxelles-Capitale    Gestion des déchets et des eaux usées   
4  Bruxelles-Capitale    Gestion des déchets et des eaux usées   

                                         activity_en  \
0  Installations for the incineration of non-haza...   
1  Installations for the incineration of non-haza...   
2  Installations for surface treatment of metals ...   
3                 Urban waste-water treatment plants   
4                 Urban waste-water treatment plants   

                                 activity_translated countryCode  \
0  Installations destinées à l'incinération des d...          BE   
1  Installations destinées à l'incinération des d...          BE   
2  Installations de traitement de surface des mét...          BE   
3  Installations de traitement des eaux urbaines ...          BE   
4  Installations de traitement des eaux urbaines ...          BE   

                                        pollutant  ...    2014 (kg)  \
0                           Nitrogen oxides (NOX)  ...     158000.0   
1                            Carbon dioxide (CO2)  ...  484000000.0   
2  Non-methane volatile organic compounds (NMVOC)  ...     322000.0   
3                             Nitrous oxide (N2O)  ...          NaN   
4                            Sulphur oxides (SOX)  ...          NaN   

     2015 (kg)    2016 (kg)    2017 (kg)    2018 (kg)    2019 (kg)  \
0     163000.0     127000.0     134000.0     106000.0     156000.0   
1  473000000.0  492000000.0  427000000.0  485000000.0  625000000.0   
2     320000.0     277000.0     244000.0     122000.0     118000.0   
3          NaN      40100.0      22100.0          NaN          NaN   
4          NaN          NaN          NaN          NaN          NaN   

     2020 (kg)    2021 (kg)    2022 (kg)  All years (kg)  
0     150000.0     124000.0      98300.0         2827300  
1  442000000.0  471000000.0  504000000.0      7442000000  
2          NaN          NaN     104000.0         3495000  
3          NaN          NaN          NaN           62200  
4          NaN     516000.0          NaN          516000  

[5 rows x 28 columns]

In [27]:
dftime = dftime.drop(["Facility_INSPIRE_ID", "activity_translated", "countryCode", "pollutant_translated"], axis='columns')

In [28]:
dftime.dtypes

facilityName          object
parentCompanyName     object
city                  object
region                object
sector                object
activity_en           object
pollutant             object
2007 (kg)            float64
2008 (kg)            float64
2009 (kg)            float64
2010 (kg)            float64
2011 (kg)            float64
2012 (kg)            float64
2013 (kg)            float64
2014 (kg)            float64
2015 (kg)            float64
2016 (kg)            float64
2017 (kg)            float64
2018 (kg)            float64
2019 (kg)            float64
2020 (kg)            float64
2021 (kg)            float64
2022 (kg)            float64
All years (kg)         int64
dtype: object

In [29]:
# Selecting only the top30 installation that polluted the most in terms of economical cost, 
# following the previous analysis

top_facs = ["ARCELORMITTAL BELGIUM - GENT", 
"TotalEnergies Refinery Antwerp", 
"AHLSTROM  MUNKSJÖ MALMEDY", 
"Electrabel - Centrale Knippegroen", 
"BASF ANTWERPEN", 
"EXXONMOBIL PETROLEUM & CHEMICAL - ESSO RAFFINADERIJ", 
"HOLCIM Belgique - Usine d'OBOURG", 
"ELECTRABEL CENTRALE RODENHUIZE", 
"CCB - Site de Gaurain-Ramecroix", 
"CBR sa - Site d'Antoing",
"CBR sa - Site de Lixhe",
"BURGO ARDENNES", 
"UMICORE - HOBOKEN",            
"AIR LIQUIDE LARGE INDUSTRY",                      
"TotalEnergies Olefins Antwerp",                       
"ELECTRABEL - CENTRALE D'AMERCOEUR",                   
"INDUSTRIE DU BOIS VIELSALM & CIE - IBV",              
"ARGEX",                                                
"EVONIK ANTWERPEN",                                
"ZANDVLIET POWER - TERREIN BASF",                        
"STORA ENSO LANGERBRUGGE",                              
"EUROCHEM ANTWERPEN",                                  
"WIENERBERGER",                                      
"BIOWANZE",                                      
"LUMINUS",                                             
"T POWER",                                            
"CARRIERES ET FOURS A CHAUX DUMONT WAUTIER",              
"LANXESS",                                               
"ELECTRABEL - CENTRALE BAUDOUR/St GHISLAIN",             
"INOVYN Manufacturing Belgium"]

In [30]:
len(top_facs)

30

In [31]:
dftimetop = dftime.loc[dftime["facilityName"].isin(top_facs)]

In [32]:
len(dftimetop)

137

### By sector

In [33]:
dftimetop["sector"].value_counts()

sector
Industrie minérale                                    44
Secteur énergétique                                   43
Industrie chimique                                    30
Production et transformation des métaux               11
Fabrication et transformation du papier et du bois     9
Name: count, dtype: int64

Creating a macro df with sectors' combined pollution in kg over the years

In [34]:
sector11 = dftimetop.groupby("sector")["2011 (kg)"].sum()

In [35]:
sector12 = dftimetop.groupby("sector")["2012 (kg)"].sum()

In [36]:
sector13 = dftimetop.groupby("sector")["2013 (kg)"].sum()

In [37]:
sector14 = dftimetop.groupby("sector")["2014 (kg)"].sum()

In [38]:
sector15 = dftimetop.groupby("sector")["2015 (kg)"].sum()

In [39]:
sector16 = dftimetop.groupby("sector")["2016 (kg)"].sum()

In [40]:
sector17 = dftimetop.groupby("sector")["2017 (kg)"].sum()

In [41]:
sector18 = dftimetop.groupby("sector")["2018 (kg)"].sum()

In [42]:
sector19 = dftimetop.groupby("sector")["2019 (kg)"].sum()

In [43]:
sector20 = dftimetop.groupby("sector")["2020 (kg)"].sum()

In [44]:
sector21 = dftimetop.groupby("sector")["2021 (kg)"].sum()

Merging sector/year dfs

In [45]:
sector_1121 = pd.merge(sector11, sector12, on="sector")

In [46]:
sector_1121 = pd.merge(sector_1121, sector13, on="sector")

In [47]:
sector_1121 = pd.merge(sector_1121, sector14, on="sector")

In [48]:
sector_1121 = pd.merge(sector_1121, sector15, on="sector")

In [49]:
sector_1121 = pd.merge(sector_1121, sector16, on="sector")

In [50]:
sector_1121 = pd.merge(sector_1121, sector17, on="sector")

In [51]:
sector_1121 = pd.merge(sector_1121, sector18, on="sector")

In [52]:
sector_1121 = pd.merge(sector_1121, sector19, on="sector")

In [53]:
sector_1121 = pd.merge(sector_1121, sector20, on="sector")

In [54]:
sector_1121 = pd.merge(sector_1121, sector21, on="sector")

In [55]:
sector_1121.to_csv("sector_1121.csv")

In [56]:
dftimetop.head()

facilityName parentCompanyName         city  \
23             WIENERBERGER      WIENERBERGER  Tessenderlo   
31             WIENERBERGER      WIENERBERGER       Beerse   
130            WIENERBERGER      WIENERBERGER       Pittem   
257  CBR sa - Site de Lixhe  Heidelbergcement        Lixhe   
258  CBR sa - Site de Lixhe  Heidelbergcement        Lixhe   

                  region              sector  \
23              Limbourg  Industrie minérale   
31                Anvers  Industrie minérale   
130  Flandre-Occidentale  Industrie minérale   
257      Région wallonne  Industrie minérale   
258      Région wallonne  Industrie minérale   

                                           activity_en              pollutant  \
23   Installations for the manufacture of ceramic p...   Sulphur oxides (SOX)   
31   Installations for the manufacture of ceramic p...   Sulphur oxides (SOX)   
130  Installations for the manufacture of ceramic p...   Sulphur oxides (SOX)   
257  Installations for the production of cement cli...  Nitrogen oxides (NOX)   
258  Installations for the production of cement cli...          Ammonia (NH3)   

     2007 (kg)  2008 (kg)  2009 (kg)  ...  2014 (kg)  2015 (kg)  2016 (kg)  \
23    710000.0   919000.0   559000.0  ...   255000.0   156000.0        NaN   
31    510000.0   321000.0   330000.0  ...        NaN        NaN        NaN   
130        NaN   374000.0   322000.0  ...        NaN        NaN        NaN   
257  2750000.0  2670000.0  2030000.0  ...  1800000.0  1670000.0  1410000.0   
258    28000.0    26100.0    35400.0  ...    49700.0    33100.0    18300.0   

     2017 (kg)  2018 (kg)  2019 (kg)  2020 (kg)  2021 (kg)  2022 (kg)  \
23         NaN        NaN        NaN        NaN        NaN        NaN   
31         NaN        NaN        NaN        NaN        NaN        NaN   
130        NaN        NaN        NaN        NaN        NaN        NaN   
257   986000.0   960000.0   966000.0  1380000.0  1180000.0  1080000.0   
258    28700.0    21000.0    21100.0    18000.0    13900.0    12700.0   

     All years (kg)  
23          3192000  
31          1549000  
130         1195000  
257        28002000  
258          435300  

[5 rows x 24 columns]

## Calculations and analysis

Creating table with total pollutants over the years for comparison per installation

In [57]:
df2021 = dftimetop.groupby("facilityName")["2021 (kg)"].sum()

In [58]:
df2016 = dftimetop.groupby("facilityName")["2016 (kg)"].sum()

In [59]:
df2011 = dftimetop.groupby("facilityName")["2011 (kg)"].sum()

In [60]:
dfyears = pd.merge(df2011, df2016, on = 'facilityName')

In [61]:
dfyears = pd.merge(dfyears, df2021, on = 'facilityName')

In [62]:
dfyears.head()

2011 (kg)     2016 (kg)     2021 (kg)
facilityName                                                          
AIR LIQUIDE LARGE INDUSTRY    7.421610e+08  5.911510e+08  1.028194e+09
ARCELORMITTAL BELGIUM - GENT  3.713661e+09  4.173847e+09  3.651097e+09
ARGEX                         1.262000e+06  1.010000e+06  1.224100e+06
BASF ANTWERPEN                2.884586e+09  3.112422e+09  2.962239e+09
BIOWANZE                      4.572460e+08  5.142210e+08  5.812269e+08

In [63]:
dfyears.shape

(28, 3)

In [64]:
dfyears.to_csv("polluters_2011_21.csv")

Adding parent company

In [65]:
df_parent = dftime[["facilityName", "parentCompanyName"]]

In [66]:
dfyears = pd.merge(left = dfyears, 
               right = df_parent, 
               how = 'left', 
               on = 'facilityName')

In [67]:
dfyears.head()

facilityName     2011 (kg)     2016 (kg)     2021 (kg)  \
0    AIR LIQUIDE LARGE INDUSTRY  7.421610e+08  5.911510e+08  1.028194e+09   
1    AIR LIQUIDE LARGE INDUSTRY  7.421610e+08  5.911510e+08  1.028194e+09   
2    AIR LIQUIDE LARGE INDUSTRY  7.421610e+08  5.911510e+08  1.028194e+09   
3  ARCELORMITTAL BELGIUM - GENT  3.713661e+09  4.173847e+09  3.651097e+09   
4  ARCELORMITTAL BELGIUM - GENT  3.713661e+09  4.173847e+09  3.651097e+09   

            parentCompanyName  
0  AIR LIQUIDE LARGE INDUSTRY  
1  AIR LIQUIDE LARGE INDUSTRY  
2  AIR LIQUIDE LARGE INDUSTRY  
3       ARCELORMITTAL BELGIUM  
4       ARCELORMITTAL BELGIUM

In [68]:
dfyears = dfyears.drop_duplicates() #All were back, so dropping

In [69]:
dfyears.shape

(28, 5)

In [70]:
dfcompany = dfyears.groupby("parentCompanyName").sum()

In [71]:
dfcompany.to_csv("companies_2011_21.csv")

## Including longlat 

In [72]:
ci = pd.read_csv("worldcities.csv")

In [73]:
ci = ci.loc[ci["iso2"] == "BE"]

In [74]:
ci.head()

city city_ascii      lat     lng  country iso2 iso3  \
619    Brussels   Brussels  50.8467  4.3525  Belgium   BE  BEL   
1381    Antwerp    Antwerp  51.2178  4.4003  Belgium   BE  BEL   
2552       Gent       Gent  51.0536  3.7253  Belgium   BE  BEL   
3273  Charleroi  Charleroi  50.4000  4.4333  Belgium   BE  BEL   
3383      Liège      Liege  50.6397  5.5706  Belgium   BE  BEL   

                   admin_name  capital  population          id  
619   Brussels-Capital Region  primary   1235192.0  1056469830  
1381                 Flanders    minor    536079.0  1056168623  
2552                 Flanders    minor    265086.0  1056062897  
3273                 Wallonia    minor    201816.0  1056263311  
3383                 Wallonia    minor    195278.0  1056513284

In [75]:
ci = ci.drop(['iso2', 'iso3', 'id'], axis=1)

In [76]:
cols = ["city", "lat", "lng", "population"]
ci_small = ci[cols]    #Keeping original cities cols

In [77]:
ci_small.head()

city      lat     lng  population
619    Brussels  50.8467  4.3525   1235192.0
1381    Antwerp  51.2178  4.4003    536079.0
2552       Gent  51.0536  3.7253    265086.0
3273  Charleroi  50.4000  4.4333    201816.0
3383      Liège  50.6397  5.5706    195278.0

## Including missing cities
I did this for the top 10, didn't continue for the top 30

In [78]:
new_rows = pd.DataFrame({
    'city': ['Antwerpen', 'Gaurain-Ramecroix', 'Obourg', 'Antoing'],
    'lat': [51.2178, 50.589, 50.476111, 50.5677],
    'lng': [4.4003, 3.488, 4.006111, 3.4492],
    'population': [536079.0, 3619.0, 4488.0, 7760.0]
})

# Adding the new rows
ci_small = pd.concat([ci_small, new_rows], ignore_index=True)


In [79]:
top = pd.merge(left = dftimetop, 
               right = ci_small, 
               how = 'left', 
               on = 'city')

In [80]:
top.sample()

facilityName parentCompanyName  city  \
118  Electrabel - Centrale Knippegroen        ELECTRABEL  Gent   

                region               sector  \
118  Flandre orientale  Secteur énergétique   

                                           activity_en             pollutant  \
118  Thermal power stations and other combustion in...  Sulphur oxides (SOX)   

     2007 (kg)  2008 (kg)  2009 (kg)  ...  2017 (kg)  2018 (kg)  2019 (kg)  \
118        NaN        NaN        NaN  ...   529000.0   341000.0   389000.0   

     2020 (kg)  2021 (kg)  2022 (kg)  All years (kg)      lat     lng  \
118   463000.0   453000.0   612000.0         5875000  51.0536  3.7253   

     population  
118    265086.0  

[1 rows x 27 columns]

In [81]:
top[top["lat"].isna()] #No latlng NaNs? OK!

facilityName  \
2                                WIENERBERGER   
3                      CBR sa - Site de Lixhe   
4                      CBR sa - Site de Lixhe   
5                      CBR sa - Site de Lixhe   
6                      CBR sa - Site de Lixhe   
7                      CBR sa - Site de Lixhe   
29  CARRIERES ET FOURS A CHAUX DUMONT WAUTIER   
30  CARRIERES ET FOURS A CHAUX DUMONT WAUTIER   
31  CARRIERES ET FOURS A CHAUX DUMONT WAUTIER   
32  CARRIERES ET FOURS A CHAUX DUMONT WAUTIER   
33               INOVYN Manufacturing Belgium   
34               INOVYN Manufacturing Belgium   
35               INOVYN Manufacturing Belgium   
36               INOVYN Manufacturing Belgium   
47  ELECTRABEL - CENTRALE BAUDOUR/St GHISLAIN   
48  ELECTRABEL - CENTRALE BAUDOUR/St GHISLAIN   
52     INDUSTRIE DU BOIS VIELSALM & CIE - IBV   
53     INDUSTRIE DU BOIS VIELSALM & CIE - IBV   
54     INDUSTRIE DU BOIS VIELSALM & CIE - IBV   
55     INDUSTRIE DU BOIS VIELSALM & CIE - IBV   
56     INDUSTRIE DU BOIS VIELSALM & CIE - IBV   
57     INDUSTRIE DU BOIS VIELSALM & CIE - IBV   
58     INDUSTRIE DU BOIS VIELSALM & CIE - IBV   

                            parentCompanyName                     city  \
2                                WIENERBERGER                   Pittem   
3                            Heidelbergcement                    Lixhe   
4                            Heidelbergcement                    Lixhe   
5                            Heidelbergcement                    Lixhe   
6                            Heidelbergcement                    Lixhe   
7                            Heidelbergcement                    Lixhe   
29  CARRIERES ET FOURS A CHAUX DUMONT-WAUTIER  Saint-Georges-Sur-Meuse   
30  CARRIERES ET FOURS A CHAUX DUMONT-WAUTIER  Saint-Georges-Sur-Meuse   
31  CARRIERES ET FOURS A CHAUX DUMONT-WAUTIER  Saint-Georges-Sur-Meuse   
32  CARRIERES ET FOURS A CHAUX DUMONT-WAUTIER  Saint-Georges-Sur-Meuse   
33               INOVYN Manufacturing Belgium       Jemeppe-Sur-Sambre   
34               INOVYN Manufacturing Belgium       Jemeppe-Sur-Sambre   
35               INOVYN Manufacturing Belgium       Jemeppe-Sur-Sambre   
36               INOVYN Manufacturing Belgium       Jemeppe-Sur-Sambre   
47                                 ELECTRABEL                  Baudour   
48                                 ELECTRABEL                  Baudour   
52              I.B.V. et Cie  ((IBV et Cie))                 Vielsalm   
53              I.B.V. et Cie  ((IBV et Cie))                 Vielsalm   
54              I.B.V. et Cie  ((IBV et Cie))                 Vielsalm   
55              I.B.V. et Cie  ((IBV et Cie))                 Vielsalm   
56              I.B.V. et Cie  ((IBV et Cie))                 Vielsalm   
57              I.B.V. et Cie  ((IBV et Cie))                 Vielsalm   
58              I.B.V. et Cie  ((IBV et Cie))                 Vielsalm   

                 region               sector  \
2   Flandre-Occidentale   Industrie minérale   
3       Région wallonne   Industrie minérale   
4       Région wallonne   Industrie minérale   
5       Région wallonne   Industrie minérale   
6       Région wallonne   Industrie minérale   
7       Région wallonne   Industrie minérale   
29      Région wallonne   Industrie minérale   
30      Région wallonne   Industrie minérale   
31      Région wallonne   Industrie minérale   
32      Région wallonne   Industrie minérale   
33    Province de Namur   Industrie chimique   
34    Province de Namur   Industrie chimique   
35    Province de Namur   Industrie chimique   
36    Province de Namur   Industrie chimique   
47           Luxembourg  Secteur énergétique   
48           Luxembourg  Secteur énergétique   
52           Luxembourg  Secteur énergétique   
53           Luxembourg  Secteur énergétique   
54           Luxembourg  Secteur énergétique   
55           Luxembourg  Secteur énergétique   
56           Luxembourg  Secteur énergétique   
57           Luxembourg

In [82]:
ci_small.to_csv("Belgian_cities.csv", index=False)

In [83]:
dfcostbe.to_csv("pollution_cost_be_per_fac.csv", index=False)